dynatrace/src/main/java/com/dynatrace/opentelemetry/metric/DynatraceMetricExporter.java

```diff
@@ -102,32 +131,113 @@ public CompletableResultCode export(@Nonnull Collection<MetricData> metrics) {
-    try {
-      connection.setRequestMethod("POST");
-      connection.setRequestProperty("Accept", "*/*; q=0");
-      connection.setRequestProperty("Authorization", "Api-Token " + apiToken);
-      connection.setRequestProperty("Content-Type", "text/plain; charset=utf-8");
-      connection.setDoOutput(true);
-      try (final OutputStream outputStream = connection.getOutputStream()) {
-        outputStream.write(mintMetricsMessage.getBytes(StandardCharsets.UTF_8));
+
+    List<String> metricLines = makeMetricLines(metrics);
+    for (List<String> partition :
+        Lists.partition(metricLines, DynatraceMetricApiConstants.getPayloadLinesLimit())) {
+      CompletableResultCode resultCode;
+      String joinedMetricLines = Joiner.on('\n').join(partition);
+
+      if (logger.isLoggable(Level.FINER)) {
+        logger.finer(String.format("Exporting metrics:\n%s", joinedMetricLines));
       }
-      int code = connection.getResponseCode();
-      if (code != 202) {
-        logger.log(Level.WARNING, "Received error code {0} from server", code);
-        return CompletableResultCode.ofFailure();
+      try {
+        connection.setRequestMethod("POST");
+        connection.setRequestProperty("Accept", "*/*; q=0");
+        if (this.apiToken != null) {
+          connection.setRequestProperty("Authorization", "Api-Token " + apiToken);
+        }
+        connection.setRequestProperty("Content-Type", "text/plain; charset=utf-8");
+        connection.setDoOutput(true);
+        try (final OutputStream outputStream = connection.getOutputStream()) {
+          outputStream.write(joinedMetricLines.getBytes(StandardCharsets.UTF_8));
+        }
+        int code = connection.getResponseCode();
+        if (code < 400) {
+          String response =
+              CharStreams.toString(
+                  new InputStreamReader(connection.getInputStream(), Charsets.UTF_8));
+          resultCode = handleSuccess(code, metricLines.size(), response);
+        } else {
+          resultCode = CompletableResultCode.ofFailure();
+        }
+      } catch (Exception e) {
+        logger.log(Level.WARNING, "Error while exporting", e);
+        resultCode = CompletableResultCode.ofFailure();
+      }
+      if (!resultCode.isSuccess()) {
+        return resultCode;
       }
-    } catch (Exception e) {
-      logger.log(Level.WARNING, "Error while exporting", e);
-      return CompletableResultCode.ofFailure();
     }
     return CompletableResultCode.ofSuccess();
   }
 
+  private CompletableResultCode handleSuccess(int code, int totalLines, String response) {
+    if (code == 202) {
+      if (RETURNED_ERROR_FIELD_IS_NULL.matcher(response).find()) {
+        Matcher linesOkMatchResult = EXTRACT_LINES_OK.matcher(response);
+        Matcher linesInvalidMatchResult = EXTRACT_LINES_INVALID.matcher(response);
+        if (linesOkMatchResult.find() && linesInvalidMatchResult.find()) {
+          logger.info(
```

Comment:
I think level FINE would be more appropriate here which should be combined with a check of isLoggable.